pandas/io/excel/_openpyxl.py

```diff
@@ -61,17 +61,23 @@ def __init__(
             self.sheets = {name: self.book[name] for name in self.book.sheetnames}
 
         else:
-            # Create workbook object with default optimized_write=True.
-            self.book = Workbook()
+            try:
+                # Sheets are not automatically created in the workbook
+                self.book = Workbook(write_only=True)
+            except ImportError:
+                print("Warning: lxml is not installed")
```

Comment:
If we want to send warning to the user, we better use proper warnings, this is not the way.

Reply:
> If we want to send warning to the user, we better use proper warnings, this is not the way.

I thought so, can I ask if there is a standard that pandas uses for passing errors to the user? I think this also brings up an interesting point; if `write_only` mode is unavailable, should that throw an error and stop the code, or just initialize a read/write workbook instead?

Reply:
Python has a `warning` module, you can grep in the code for samples.

My understanding is that we are currently using unnecessary memory when reading Excel files. I think what we want is to try to save memory, but if it's not possible continue to open the file anyway. Warning the user that could be saving memory in the operation by installing a library seems reasonable. But I don't think it makes sense to fail if the user has enough memory for the operation.

Reply:
Your understanding is correct! I agree, a warning seems reasonable enough. I'm mainly asking because I'm new to contributing to pandas and still trying to get a feel for best practices within the library :)